pykokkos/interface/type_inference.py

```diff
@@ -0,0 +1,279 @@
+
+        # Check policy and apply annotation(s)
+        if isinstance(policy, RangePolicy) or isinstance(policy, TeamThreadRange):
+            # only expects one param
+            if i == 0:
+                updated_types.inferred_types[param.name] = "int"
+                updated_types.is_arg.add(param.name)
+
+        elif isinstance(policy, TeamPolicy):
+            if i == 0:
+                updated_types.inferred_types[param.name] = 'TeamMember'
+                updated_types.is_arg.add(param.name)
+
+        elif isinstance(policy, MDRangePolicy):
+            total_dims = len(policy.begin) 
+            if i < total_dims:
+                updated_types.inferred_types[param.name] = "int"
+                updated_types.is_arg.add(param.name)
+        else:
+            raise ValueError("Automatic annotations not supported for this policy")
+
+        # last policy param for parallel reduce and second last for parallel_scan is always the accumulator; the default type is double
+        if i == policy_params - 1 and parallel_type == "parallel_reduce" or i == policy_params - 2 and parallel_type == "parallel_scan":
+            updated_types.inferred_types[param.name] = "Acc:double"
+            updated_types.is_arg.add(param.name)
+
+        if i == policy_params - 1 and parallel_type == "parallel_scan":
+            updated_types.inferred_types[param.name] = "bool"
+            updated_types.is_arg.add(param.name)
+
+    return updated_types
+
+
+def infer_other_args(
+    param_list: List[inspect.Parameter], 
+    policy_params: int,
+    args_list: List[Any],
+    start_idx: int,
+    updated_types: UpdatedTypes
+    ) -> UpdatedTypes:
+    '''
+    Infer the types of arguments (after the policy arguments)
+
+    param_list: list of parameter objects that are present in the workunit signature
+    policy_params: the number of initial parameters that are dedicated to policy (in param_list/signature)
+    args_list: List of arguments passed to the parallel dispactch (e.g args for parallal_for())
+    start_idx: The index for the first non policy argument in args_list
+    updated_types: UpdatedTypes object to store inferred types information
+    returns: Updated UpdatedTypes object with inferred types
+    '''
+
+    # DataType class has all supported pk datatypes, we ignore class members starting with __
+    supported_np_dtypes = [attr for attr in dir(DataType) if not attr.startswith("__")]
+
+    for i in range(policy_params , len(param_list)):
+        param = param_list[i]
+        value = args_list[start_idx + i - policy_params]
+
+        if isinstance(value, View) and value.layout != Layout.LayoutDefault:
+            updated_types.layout_change[param.name] = "LayoutRight" if value.layout == Layout.LayoutRight else "LayoutLeft"
+
+        if param.annotation is not inspect._empty:
+            continue
+
+        param_type = type(value).__name__
+
+        # switch integer values over 31 bits (signed positive value) to pk.int64
```

Comment:
we say pk.int64 but below we use numpy:int64

Reply:
fixed